src/PricingManager/Rule.php

```diff
@@ -120,7 +120,7 @@ public function setValue(string $key, mixed $value, bool $ignoreEmptyValues = fa
                         return $this;
                     }
             }
-            $this->$method($value);
+            $this->$method($value ?? '');
```

Comment:
But we can't be sure it's always of type `string`, right?

Reply:
Will look into that.

Reply:
It seems that `setactions`
https://github.com/pimcore/ecommerce-framework-bundle/blob/f1b8a2ebcfe5467856a94c88882635aab2ea2e3b/src/PricingManager/RuleInterface.php#L49 and `setcondition`
https://github.com/pimcore/ecommerce-framework-bundle/blob/f1b8a2ebcfe5467856a94c88882635aab2ea2e3b/src/PricingManager/RuleInterface.php#L45 wouldn't work with strings

Reply:
@Daniel-Ateles any updates? :+1:

Reply:
@Daniel-Ateles friendly reminder :alarm_clock: :lollipop:

Reply:
@Corepex I guess this PR solved faulty DB-data in my case and should have been solved by removing the faulty rules from the db table instead. This PR can be closed/removed.